File: client/lib/models/message.dart
class MessageModel {
  final int id;
  final int senderId;
  final int receiverId;
  final String? content;
  final String msgType; // text, image, voice, video
  final String? fileUrl;
  final DateTime createdAt;
  final String? senderName;

  MessageModel({
    required this.id,
    required this.senderId,
    required this.receiverId,
    this.content,
    required this.msgType,
    this.fileUrl,
    required this.createdAt,
    this.senderName,
  });

  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
      id: json['id'],
      senderId: json['sender_id'],
      receiverId: json['receiver_id'],
      content: json['content'],
      msgType: json['msg_type'] ?? 'text',
      fileUrl: json['file_url'],
      createdAt: DateTime.parse(json['created_at']),
      senderName: json['sender_name'],
    );
  }

  bool get isImage => msgType == 'image';
  bool get isVoice => msgType == 'voice';
  bool get isVideo => msgType == 'video';
  bool get isText => msgType == 'text';
}

File: backend/app/models/message.py
import datetime
from sqlalchemy import String, Integer, DateTime, ForeignKey, func, Enum as SAEnum
from sqlalchemy.orm import Mapped, mapped_column, relationship
import enum

from app.database import Base


class MessageType(str, enum.Enum):
    TEXT = "text"
    IMAGE = "image"
    VOICE = "voice"
    VIDEO = "video"
    SYSTEM = "system"


class Message(Base):
    __tablename__ = "messages"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    sender_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id"), nullable=False
    )
    receiver_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id"), nullable=False
    )
    content: Mapped[str | None] = mapped_column(String(2000), default=None)
    msg_type: Mapped[MessageType] = mapped_column(
        SAEnum(MessageType), default=MessageType.TEXT
    )
    file_url: Mapped[str | None] = mapped_column(String(500), default=None)
    created_at: Mapped[datetime.datetime] = mapped_column(
        DateTime, server_default=func.now()
    )

    sender = relationship("User", back_populates="sent_messages", foreign_keys=[sender_id])
